packages/teams-js/src/public/media.ts

```diff
@@ -412,30 +414,47 @@ export namespace media {
      * @param mediaEvent indicates what the event that needs to be signaled to the host client
      * Optional; @param callback is used to send app if host client has successfully handled the notification event or not
      */
-    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void {
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
       ensureInitialized(FrameContexts.content, FrameContexts.task);
-      const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
-      if (err) {
-        if (callback) {
-          callback(err);
-        }
-        return;
-      }
+      let mediaError;
+      const promise = Promise.resolve().then(() =>
+        throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion),
+      );
 
       const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+
       sendMessageToParent('media.controller', [params], (err?: SdkError) => {
         if (callback) {
-          callback(err);
+          return callback(err);
+        } else if (err) {
+          mediaError = err;
+        }
+      });
+
+      return promise.then(() => {
+        if (mediaError) {
+          throw mediaError;
         }
       });
     }
 
     /**
+     * Function to programatically stop the ongoing media event
+     *
+     * @returns A resolved promise
+     * */
+    public stop(): Promise<void>;
+    /**
+     * @deprecated
+     * As of 2.0.0-beta.3, please use {@link MediaController.stop MediaController.stop(): Promise\<void\>} instead.
+     
      * Function to programatically stop the ongoing media event
      * Optional; @param callback is used to send app if host client has successfully stopped the event or not
      */
-    public stop(callback?: (err?: SdkError) => void): void {
-      this.notifyEventToHost(MediaControllerEvent.StopRecording, callback);
+    public stop(callback: (err?: SdkError) => void): void;
+    public stop(callback?: (err?: SdkError) => void): Promise<void> {
+      const wrappedFunction = (): Promise<void> => this.notifyEventToHost(MediaControllerEvent.StopRecording, callback);
+      return callCallbackWithErrorOrResultFromPromiseAndReturnPromise(wrappedFunction, callback);
```

Comment:
This implementation is wrong. you are calling the callback once in the notifyEventfunction, then again in the utility function.
Also why bother wrapping up with the utility function, which is btw wrong, it should be `callCallbackWithSdkErrorFromPromiseAndReturnPromise`

Just return the resolved Promise.
```
public stop(callback?: (err?: SdkError) => void): Promise<void> {
      return Promise.resolve(this.notifyEventToHost(MediaControllerEvent.StopRecording, callback));
    }
```